lib/elixir_analyzer/exercise_test/check_source/compiler.ex

```diff
@@ -20,10 +20,21 @@ defmodule ElixirAnalyzer.ExerciseTest.CheckSource.Compiler do
 
     quote do
       (fn %Source{} = source ->
-         if unquote(check_function).(source) do
-           {:pass, unquote(test_description)}
-         else
-           {:fail, unquote(test_description)}
+         case unquote(check_function).(source) do
+           true ->
+             {:pass, unquote(test_description)}
+
+           false ->
+             {:fail, unquote(test_description)}
+
+           {true, params} when is_map(params) ->
+             {:pass, %{unquote(test_description) | params: params}}
```

Comment:
I wasn't sure if this case makes sense or how to test it - if the check passes, the results don't include the comment, right? So it doesn't matter what its params are?

Reply:
Yeah that's true.
If we can't test it, it's basically dead code, so we can remove it until we have a use case for it (and therefore a test).

Reply:
If I remove `| params: params` from this line, I get quite a lot of errors. If I remove the whole two lines, same. The only change that keeps the tests passing would be to say `params: %{}` but I don't see how that's better than keeping `params: params`, so I'm not touching this code 🤷